src/søknad/steg/4-barnasbosted/BarnasBostedInnhold.tsx

```diff
@@ -72,6 +76,34 @@ const BarnasBostedInnhold: React.FC<Props> = ({
         barnMedLevendeMedforelder.length
     );
   }, [søknad]);
+  const forelderIdenterMedBarn = forelderidentMedBarn(
+    barnMedLevendeMedforelder
+  );
+
+  const oppdaterBarn = (
+    oppdatertBarn: IBarn,
+    erFørsteAvFlereBarnMedSammeForelder: boolean
+  ) => {
+    const barnMedSammeForelder =
+      oppdatertBarn.forelder?.ident?.verdi &&
+      forelderIdenterMedBarn.get(oppdatertBarn.forelder?.ident?.verdi);
+
+    if (barnMedSammeForelder && erFørsteAvFlereBarnMedSammeForelder) {
+      oppdaterFlereBarnISoknaden(
+        barnMedSammeForelder.map((b) => {
+          if (b.id === oppdatertBarn.id) {
+            return oppdatertBarn;
+          }
+
+          return oppdatertBarn.forelder
+            ? kopierFellesForeldreInformasjon(b, oppdatertBarn.forelder)
+            : b;
```

Comment:
Mulig jeg misforstår, men skulle tro at dersom b.id === oppdatertbarn.id så ville vi kopiere forelderinfo..?

Reply:
Listen `barnMedSammeForelder` inneholder det som er tidligere lagret. Dersom man gjør endringer på et barn som vil påvirke andre barn ønsker man å legge inn informasjonen som ligger i `oppdatertBarn` både for en selv (`b.id === oppdatertbarn.id`) og for resterende barn med samme "annen forelder"

Reply:
Eller mener du at det bør lages en kopi for å ikke returnere det faktiske objektet som brukes ellers? For det er en annen sak 😅

Reply:
Kortere: kopi brukes dersom man ønsker å kopiere spesifikke verdier på foreldre-objektet fordi det settes via et annet barn. Så når vi oppdaterer barn1 (nye verdier ligger i oppdatertBarn) må barn2+ få visse verdier fra oppdatertBarn